### PR TITLE
doc: Bluetooth: Mesh: Added new Sensor API migration doc

### PR DESCRIPTION
**Pull request by Balaklaka**
Added migration description on how to do migration from the deprecated Sensor API to the new Sensor API.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:657]: I don't know if we should do this now or after the docs are updated, but `bt_mesh_sensors_readme` should also be linked here since that contains the detailed description of how all the API pieces fit together (or, it will do, once it's updated)

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:655]:
```suggestion
* For Mesh applications, the Sensor API has been re-written, and the previous Sensor API is deprecated.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:658]: I'd remove this entire sentence. It's incorrect as it is, it could be rewritten to "It is important to use the new Sensor API as it can provide better precision in some use cases", but then it doesn't have much value ("why is it then important if it doesn't provide any value in my use case?"). The changelog already mentions increased precision, those who don't need that for their use case won't use that as an argument for using the new API.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:661]:
```suggestion
  This means for existing applications there is no need to do anything to continue using the deprecated Sensor API.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:663]:
```suggestion
  This will work until the deprecation period is over for the deprecated Sensor API.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:664]:
```suggestion
  When the deprecation period is over the deprecated Sensor API will be removed, and the application will no longer compile if not updated to the new Sensor API.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:666]:
```suggestion
  New and existing applications that will use the new Sensor APIs must set :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` to `=n` in their configuration.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:668]:
```suggestion
  It is important to check if :kconfig:option:`CONFIG_FPU` is set to `=y` to support accellerated floating point operations, and possible add :kconfig:option:`CONFIG_CBPRINTF_FP_SUPPORT` to support floating point printing.
```

(floating point operations will still work with `CONFIG_FPU=n`, but they will be slow)

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:669]: Not correct. Application variables can be whatever the application wants, even `sensor_value`.

Suggested rewrite: 
```suggestion
  Sensor API arguments and callback parameters previously defined as :c:struct:`sensor_value` now use :c:struct:`bt_mesh_sensor_value` instead.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:672]: This is an implementation detail and we should not mention it here, it will only cause people to start messing with `cb` directly which is something we specifically _don't_ want. 

Suggested rewording:
```suggestion
  :c:struct:`bt_mesh_sensor_value`s with a valid `format` can be converted to and from integer, float and `struct sensor_value` through `bt_mesh_sensor_value_to/from*` functions.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:671]: Mention that the correct format to use when filling in a sensor value for a channel can be found through `sensor->type.channels[i].format` for a given `struct bt_mesh_sensor sensor`

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:670]:
```suggestion
  The :c:struct:`bt_mesh_sensor_value`.format needs to be filled by the application for variables passed to the API.
```

**Review by ludvigsj — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (PavelVPV) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:668]: This ideally should be mentioned on the sensor API doc page.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (PavelVPV) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:660]: Should we tell where to find documentation for the old API?

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (PavelVPV) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:668]: Added as a comment to NCSDK-23375

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:660]: People that has used the old API knows how it is working already, and can use the old documentation. The just want confirmation on what they have is working. The people other type of people want to know about the new API. And sometime is a good that it is hard to find API that we do not want them to use, and easy to find API we want them to use.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:668]: I agree that it should be mention in the sensor API doc page, but I still think it should be here to. It can be easy to miss in the other documentation and the people that it is reading may already have their code and just want to quickly update their code, without reading to much of the new sensor API doc.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:723]: I would rather have it like the way it is. Because that matches what is done in samples. That way people can actual match this up against what was done for our sample to see the difference. It then does becomes more real.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:670]: I actual thinks the "Applicatioin variables" should be kept. The user should remember to check his/her application variables, if they should be changed to.

Comment (ludvigsj) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:670]: Application variables typically should never be defined using `bt_mesh_sensor_value`. They should be `sensor_value`, `float`, whatever their actual use case/sensor driver provides/expects. So if you want to mention this, then you have to reword it, because "uses bt_mesh_sensor_value_instead" is definitely wrong for application variables.

Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:670]: Then it is wrong in our own sample https://github.com/nrfconnect/sdk-nrf/blob/20c55d105e83c24746ed9102f846b4d9c76d3391/samples/bluetooth/mesh/sensor_server/src/model_handler.c#L63C1-L69C1

Comment (ludvigsj) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:670]: Settings are the exception I think, but if you take a look at the lines immediately below you see that `gain` and ambient light value for example are `float` now (which was my point here). But I see your point that it should be mentioned, what about this then?

```suggestion
  * Sensor API arguments, callback parameters and relevant application variables previously defined with :c:struct:`sensor_value` now use :c:struct:`bt_mesh_sensor_value` instead.
```

**Review by ludvigsj — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Comment by ludvigsj** [comment]
@Balaklaka please rebase this so it can be merged

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:682]:
```suggestion
* For the Bluetooth Mesh samples and applications, a new sensor API (see :ref:`bt_mesh_sensors_readme`) is introduced with |NCS| v2.6.0.
  The previous sensor API is deprecated.
   
  The usage of the new sensor API is demonstrated in samples :ref:`bluetooth_mesh_sensor_client`, :ref:`bluetooth_mesh_sensor_server` and :ref:`bluetooth_mesh_light_lc`.
```

**Comment by Balaklaka** [comment]
> @Balaklaka please rebase this so it can be merged

Will do, but I still need review from @mia-ko before it can be merged..

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:684]:
```suggestion
  The Kconfig option :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` is enabled by default in the deprecation period.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:688]:
```suggestion
  This means that the existing samples and applications can continue using the deprecated sensor API as normal during the deprecation period, without the additional configuration.
  The samples and applications will get a deprecation warning when compiled, that the user can choose to disregard.
  
  When the deprecation period is over, the deprecated sensor API will be removed, and the samples and applications will no longer compile unless updated to the new sensor API.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:692]:
```suggestion
  To use the new sensor API for new and existing samples and applications, disable the Kconfig option :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE` in the configuration.
  This configuration option will be removed when the deprecation period is over, and then it has to be removed from the sample and application configuration.
  It is also recommended to enable the Kconfig option :kconfig:option:`CONFIG_FPU` to support the accelerated floating point operations, and the Kconfig option :kconfig:option:`CONFIG_CBPRINTF_FP_SUPPORT` to support the  floating point printing.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:694]:
```suggestion
  * Sensor API arguments and callback parameters previously defined with :c:struct:`sensor_value` now use :c:struct:`bt_mesh_sensor_value` instead.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:695]: Is this correct, the struct?

Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:695]:
```suggestion
  * The :c:struct:`bt_mesh_sensor_value.format` needs to be filled by the application for variables passed to the sensor API.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:695]: I think this needs to be:
```suggestion
  * :c:member:`bt_mesh_sensor_value.format` needs to be filled by the application for variables passed to the Sensor API.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:699]: Should :c:struct:`bt_mesh_sensor_format` references also be :c:member:`bt_mesh_sensor_value.format`? Or is this something else?

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:696]:
```suggestion
  * There are several different types of :c:struct:`bt_mesh_sensor_format` described in the file :file:`include/bluetooth/mesh/sensor_types.h`.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:697]:
```suggestion
  * When filling in sensor values for a channel, the format can be found through ``sensor->type.channels[i].format`` defined for the given :c:struct:`bt_mesh_sensor` sensor.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:698]:
```suggestion
  * :c:struct:`bt_mesh_sensor_format` with a valid format can be converted to and from integer, float and :c:struct:`sensor_value` through ``bt_mesh_sensor_value_to/from*`` functions.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:699]:
```suggestion
  * Where the applications previously just added values directly to :c:struct:`sensor_value.val1` and :c:struct:`sensor_value.val2`, the correct way now is to use the ``bt_mesh_sensor_value_to/from*`` functions to either set or extract the values.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:701]:
```suggestion
  Example of changes that need to be done for a sensor using sensor values from the file :file:`drivers/sensor.h`:
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:730]:
```suggestion
  Example of changes that need to be done for a sensor using the float values:
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:698]: Actually, there is something weird here, I think we mean
```suggestion
  * :c:struct:`bt_mesh_sensor_value` with a valid format can be converted to and from integer, float and :c:struct:`sensor_value` through ``bt_mesh_sensor_value_to/from*`` functions.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (ludvigsj) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:699]: `bt_mesh_sensor_value.format` is `* struct bt_mesh_sensor_format`, so one is the struct itself, the other is the field on the `bt_mesh_sensor_value`. So what to reference depends on what you want to talk about (sensor values or sensor formats)

Comment (ludvigsj) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:699]: Isn't it `:c:member:`?
```suggestion
  * Where the applications previously just added values directly to :c:member:`sensor_value.val1` and :c:member:`sensor_value.val2`, the correct way now is to use the ``bt_mesh_sensor_value_to/from*`` functions to either set or extract the values.
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:692]: I have a question. Is it correct to add "samples and applications" instead of just "applications". The samples using sensor API are updated, and the users should not migrate samples. So why add samples?

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst
Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:699]: In line 698 it is correct with the struct bt_mesh_sensor_format, they are different.

Comment (Balaklaka) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst]: I saw the same, and fixed it.

**Review by mia-ko — APPROVED** [review]

**Pull request merged**
